### PR TITLE
Filter should always be sent as a list

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
Filter should always be sent as a list. We were allowing str inputs which will not work.

### CODE REVIEW

File: modelon/impact/client/options.py [hunk 1]
```diff
@@ -65,26 +65,26 @@ class SolverOptions(BaseExecutionOptions):
...
         Example::
 
             sim_opts = custom_function.get_simulation_options().with_result_filter(
                 pattern = ["*.phi"])
 
         """
-        if not isinstance(pattern, str):
-            pattern = str(pattern)
-        return self.with_values(filter=pattern)
+        if not isinstance(pattern, list):
```
Comment (iakovn) [inline modelon/impact/client/options.py:85]: Let's do it all the way: `not isinstance(pattern, list) or [p for p in pattern if not isinstance(p, str)]` and also add test

**Review by iakovn — CHANGES_REQUESTED** [review]
Good improvement! 👍 Let's do it all the way

File: modelon/impact/client/options.py (hunk 1 above)
Comment (abhilash-kumar-nair) [reply modelon/impact/client/options.py]: Thanks @iakovn ! Fixed it :)

**Review by iakovn — APPROVED** [review]
👍

**Pull request merged**